### PR TITLE
Issue 876 - Postgres SQL Automation

### PR DESCRIPTION
**Pull request by prb112**
Issue 876 - Postgres SQL Automation

### CODE REVIEW

File: build/persistence/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        java: [ 'openjdk11' ]
+        persistence: [ 'postgres' ]
+    fail-fast: false
+```
+
+Each persistence layer that is tested as part of the framework uses the default build files and the files that match the persistence name added to the `persistence.yml`.
+
+|Filename|Purpose|
+|----------|----------------|
+|bin/gather-logs.sh|Gathers the logs for the build, no|
```
Comment (lmsurpre) [inline build/persistence/README.md:28]: no?

File: fhir-server/liberty-config/config/default/fhir-server-config-postgresql.json [hunk 2]
```diff
@@ -30,10 +30,10 @@
                 "default": {
                     "type": "postgresql",
                     "connectionProperties": {
-                        "serverName": "localhost",
+                        "serverName": "postgres_postgres_1",
                         "portNumber": 5432,
-                        "databaseName": "fhirdata",
-                        "user": "fhirserver",
+                        "databaseName": "fhirdb",
+                        "user": "fhiradmin",
```
Comment (lmsurpre) [inline fhir-server/liberty-config/config/default/fhir-server-config-postgresql.json:36]: as per albert's comment, this doesn't seem to match the user created  `build/persistence/postgres/resources/db.sql` ...how does it work?

File: build/persistence/postgres/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+EXPOSE 5432
+
+# Load Build Information
+COPY --chown=postgres:postgres resources/db.sql /docker-entrypoint-initdb.d/db.sql
+COPY --chown=postgres:postgres resources/docker-entrypoint.sh /docker-entrypoint.sh
+RUN chmod 755 /docker-entrypoint.sh
+
+# Add the fhiradmin user
+RUN addgroup -S fhir && adduser -S fhiradmin -G fhir
+RUN echo "fhiradmin:fhir" | chpasswd
```
Comment (lmsurpre) [inline build/persistence/postgres/Dockerfile:30]: does this set a password?

File: fhir-server/liberty-config/config/default/fhir-server-config-postgresql.json (hunk 2 above)
Comment (lmsurpre) [reply fhir-server/liberty-config/config/default/fhir-server-config-postgresql.json:36]: now I see that it:
1. creates this system user in the dockerfile; and
2. creates this fhiradmin db user as the FHIRDB owner in the entry-point https://github.com/IBM/FHIR/pull/1234/files#diff-6d0acfaa35776bfcbc9f9cec15f60f6dR40

do the users created in db.sql still work?

Comment (prb112) [reply fhir-server/liberty-config/config/default/fhir-server-config-postgresql.json:36]: yeah - they still work.  I'll clear this up later on, but great points, I only meant FHIRADMIN to deploy the schema, and fhirserver was meant to test/run it

File: build/persistence/postgres/Dockerfile (hunk 3 above)
Comment (prb112) [reply build/persistence/postgres/Dockerfile:30]: yeah - to fhir, I'll fix this

Comment (prb112) [reply build/persistence/postgres/Dockerfile:30]: So it uses chpasswd that's available in alpine examples... 
https://linoxide.com/linux-command/change-passwords-batch-mode-chpasswd/

Comment (prb112) [reply build/persistence/postgres/Dockerfile:30]: I actually removed this.

File: fhir-server/liberty-config/config/default/fhir-server-config-postgresql.json (hunk 2 above)
Comment (prb112) [reply fhir-server/liberty-config/config/default/fhir-server-config-postgresql.json]: I'm trying this out.  I had meant to keep fhirserver.  I'm going to check/test this, and update the Dockerfile to reflect it.

Comment (prb112) [reply fhir-server/liberty-config/config/default/fhir-server-config-postgresql.json]: Cleaned this up so we create the users as much as possible in the dockerfile.

File: build/persistence/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+|bin/pre-integration-test.sh|Call the pre-integration-test step for `persistence`|
+|persistence/pre-integration-test.sh|Run before integration-test.sh to startup image and services for the integration testing|
+|bin/post-integration-test.sh|Call the post-integration-test step for `persistence`|
+|persistence/post-integration-test.sh|Run after integration-test.sh to stop image and services from the integration testing|
+|persistence/Dockerfile|The Docker file used in development and end-to-end tests|
+|persistence/docker-compose.yml|The Docker Compose file used with the end-to-end tests|
+|persistence/resources| Stores files used to support the end-to-end tests. |
+|persistence/README.md|Describes the relevant details for this persistence layer.|
+|persistence/.gitignore|Ignores files related to the persistence layer's tests|
+|`README.md`|This file describing the framework|
```
Comment (lmsurpre) [inline build/persistence/README.md:40]: should "persistence" be "<persistence" (or similar) to indicate that is a placeholder for a specific persistence impl (postgres in this case)?

Comment (prb112) [reply build/persistence/README.md:40]: Fixed

**Pull request merged**